include/ginkgo/core/matrix/dense.hpp

```diff
@@ -1107,6 +1129,7 @@ class Dense
 private:
     Array<value_type> values_;
     size_type stride_;
+    strategy_type strategy_{strategy_type::gko};
```

Comment:
Are you sure we want to have only one strategy type for the entire class `Dense`? What if someone wants gko norm2 and vendor dot? Or we might add the option of vendor apply (once Vector is separated from Dense), and then that leads to even more options. Unless there's a better solution, I'd suggest
```suggestion
    strategy_type dot_strategy_{strategy_type::gko};
    strategy_type norm2_strategy_{strategy_type::gko};
```
and so on.